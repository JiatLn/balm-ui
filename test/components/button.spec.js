import { shallowMount } from '@vue/test-utils';
import UiButton from '@/components/button/button.vue';

describe('<ui-button>', () => {
  it('renders default button', () => {
    const wrapper = shallowMount(UiButton);

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.element.tagName).toBe('BUTTON');
    expect(wrapper.classes()).toContain('mdc-button');
  });

  it('renders outlined button', () => {
    const wrapper = shallowMount(UiButton, {
      propsData: {
        outlined: true
      }
    });

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.classes()).toContain('mdc-button--outlined');
  });

  it('renders raised button', () => {
    const wrapper = shallowMount(UiButton, {
      propsData: {
        raised: true
      }
    });

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.classes()).toContain('mdc-button--raised');
  });

  it('renders unelevated button', () => {
    const wrapper = shallowMount(UiButton, {
      propsData: {
        unelevated: true
      }
    });

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.classes()).toContain('mdc-button--unelevated');
  });

  it('renders button with icon', () => {
    const icon = 'favorite';
    const wrapper = shallowMount(UiButton, {
      propsData: {
        icon
      }
    });

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.find('.mdc-button__icon').text()).toBe(icon);
  });

  it('renders button with slot-scope', () => {
    const wrapper = shallowMount(UiButton, {
      scopedSlots: {
        before: '<i slot-scope="{ iconClass }">{{ iconClass }}</i>'
      }
    });

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.text()).toBe('mdc-button__icon');
  });

  it('renders button with @click', async () => {
    const wrapper = shallowMount(UiButton);

    await wrapper.trigger('click');

    expect(wrapper.emitted().click).toBeTruthy();
  });
});
