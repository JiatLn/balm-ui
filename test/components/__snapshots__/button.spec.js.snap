// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ui-button> renders button with icon 1`] = `
<button
  class="mdc-button"
  type="button"
>
  <div
    class="mdc-button__ripple"
  />
  <!-- Icon (optional) -->
  
  <i
    aria-hidden="true"
    class="material-icons mdc-button__icon"
  >
    favorite
  </i>
  
  <!-- Text label -->
  <span
    class="mdc-button__label"
  >
    
    
  </span>
  <!-- Icon (optional) -->
  
  
</button>
`;

exports[`<ui-button> renders button with scoped slots 1`] = `
<button
  class="mdc-button"
  type="button"
>
  <div
    class="mdc-button__ripple"
  />
  <!-- Icon (optional) -->
  
  
  <i
    class="mdc-button__icon"
  >
    mdc-button__icon
  </i>
  
  
  <!-- Text label -->
  <span
    class="mdc-button__label"
  >
    
    
  </span>
  <!-- Icon (optional) -->
  
  
</button>
`;

exports[`<ui-button> renders default button 1`] = `
<button
  class="mdc-button"
  type="button"
>
  <div
    class="mdc-button__ripple"
  />
  <!-- Icon (optional) -->
  
  <!--v-if-->
  
  <!-- Text label -->
  <span
    class="mdc-button__label"
  >
    
    
  </span>
  <!-- Icon (optional) -->
  
  
</button>
`;

exports[`<ui-button> renders outlined button 1`] = `
<button
  class="mdc-button mdc-button--outlined"
  type="button"
>
  <div
    class="mdc-button__ripple"
  />
  <!-- Icon (optional) -->
  
  <!--v-if-->
  
  <!-- Text label -->
  <span
    class="mdc-button__label"
  >
    
    
  </span>
  <!-- Icon (optional) -->
  
  
</button>
`;

exports[`<ui-button> renders raised button 1`] = `
<button
  class="mdc-button mdc-button--raised"
  type="button"
>
  <div
    class="mdc-button__ripple"
  />
  <!-- Icon (optional) -->
  
  <!--v-if-->
  
  <!-- Text label -->
  <span
    class="mdc-button__label"
  >
    
    
  </span>
  <!-- Icon (optional) -->
  
  
</button>
`;

exports[`<ui-button> renders unelevated button 1`] = `
<button
  class="mdc-button mdc-button--unelevated"
  type="button"
>
  <div
    class="mdc-button__ripple"
  />
  <!-- Icon (optional) -->
  
  <!--v-if-->
  
  <!-- Text label -->
  <span
    class="mdc-button__label"
  >
    
    
  </span>
  <!-- Icon (optional) -->
  
  
</button>
`;
